include/ic_components.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "ic_graphics.h"
+
+#include <glm/glm.hpp>
+#include <imgui.h>
+
+#include <vector>
+
+namespace IC {
+    class GameObjectComponent {
+    public:
+        GameObjectComponent();
+        ~GameObjectComponent();
+
+        virtual void Gui() = 0;
+
+    protected:
+    };
+
+    class Transform : public GameObjectComponent {
+    public:
+        Transform();
+        ~Transform();
+
+        glm::vec3 Position() { return _position; }
+        glm::vec3 Rotation() { return _rotation; }
+        glm::vec3 Scale() { return _scale; }
```

Comment:
Return all these by reference (or even const reference) so we're not making copies everytime we want to access this data.

Reply:
Honestly, could just make the three variables public since the getters & setters are default anyway.